Features/Inventory/BigInventory.cs

```diff
@@ -0,0 +1,365 @@
+	{
+		return _inventories[item.Category].Contains(item, amount);
+	}
+
+	public bool ContainsAtLeast(ItemStack item, int amount)
+	{
+		return _inventories[item.Category].ContainsAtLeast(item, amount);
+	}
+
+	public Dictionary<int, ItemStack> All(string itemId)
+	{
+		var all = new Dictionary<int, ItemStack>();
+		var index = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var items = inventory.All(itemId);
+			foreach (var (key, value) in items)
+			{
+				all.Add(key + index, value);
+			}
+
+			index += inventory.Size;
+		}
+
+		return all;
+	}
+
+	public Dictionary<int, ItemStack> All(ItemStack item)
+	{
+		var indexOffset = 0;
+		var results = new Dictionary<int, ItemStack>();
+		foreach (var (category, inventory) in _inventories)
+		{
+			// Technically redundant since wrongInventory.First() will return empty dictionary
+			if (category != item.Category)
+			{
+				indexOffset += inventory.Size;
+			}
+
+			// Get local indices and add offset for "global" index
+			var indices = inventory.All(item);
+			foreach (var (stackIndex, stack) in indices)
+			{
+				results.Add(stackIndex + indexOffset, stack);
+			}
+		}
+
+		return results;
+	}
+
+	public int First(string itemId)
+	{
+		var indexOffset = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var first = inventory.First(itemId);
+			if (first != -1)
+			{
+				return first + indexOffset;
+			}
+
+			indexOffset += inventory.Size;
+		}
+
+		return -1;
+	}
+
+	public int First(ItemStack item)
+	{
+		var indexOffset = 0;
+		foreach (var (category, inventory) in _inventories)
+		{
+			// Technically redundant since wrongInventory.First() will return -1
+			if (category != item.Category)
+			{
+				indexOffset += inventory.Size;
+				continue;
+			}
+
+			var first = inventory.First(item);
+			if (first != -1)
+			{
+				return first + indexOffset;
+			}
+		}
+
+		return -1;
+	}
+
+	public int FirstEmpty()
+	{
+		// I see no use for this
+		// ~ kida31
+		var indexOffset = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var firstEmpty = inventory.FirstEmpty();
+			if (firstEmpty != -1)
+			{
+				return firstEmpty + indexOffset;
+			}
+
+			indexOffset += inventory.Size;
+		}
+
+		return -1;
+	}
+
+	public void Remove(string itemId)
+	{
+		foreach (var inventory in _inventories.Values)
+		{
+			inventory.Remove(itemId);
+		}
+	}
+
+	public void Remove(ItemStack item)
+	{
+		foreach (var inventory in _inventories.Values)
+		{
+			inventory.Remove(item);
+		}
+	}
+
+	public void Clear(int index)
+	{
+		foreach (var inventory in _inventories.Values)
+		{
+			if (index < inventory.Size)
+			{
+				inventory.Clear(index);
+				return;
+			}
+
+			index -= inventory.Size;
+		}
+	}
+
+	public void Clear()
+	{
+		foreach (var inventory in _inventories.Values)
+		{
+			inventory.Clear();
+		}
+	}
+
+	public string GetTitle()
+	{
+		return Name;
+	}
+
+	public Dictionary<int, ItemStack> GetItemsOfCategory(ItemCategory category)
```

Comment:
Is this just the specific Version of GetContents()? 
If so rename one of those to match name

Reply:
Renamed to `GetContentOfCategory`